.circleci/config.yml

```diff
@@ -96,7 +94,8 @@ workflows:
       - nuke_phx_devops:
           requires:
             - test
-
+          context:
```

Comment:
There is a `test` job defined above, on line 95, that also needs the `context`, but it doesn't look like it was added? Perhaps because `test` has no child nodes?

Reply:
I see what you mean and I can repro this locally with YQ.

Reply:
Yes - I believe you are correct. After fiddling with this for a while, I think what's happening is that the [path expression](https://mikefarah.gitbook.io/yq/usage/path-expressions) I'm currently using assumes that jobs found under `workflows` will be either an array or a map - and the syntax supports accessing either of those data types and appending to (or creating) the `context` array that should be there. 

I think I may need a separate query that detects these single string jobs and replaces them with maps of the same name and containing a context key - in a second pass.

Reply:
This also *might* fall under the category of small annoyances we'll need to tolerate for the first pass and fix manually - I can't currently see a path to selecting or updating nodes by their type in `yq` docs.

Reply:
Ah, I see. I have two ideas for you to consider:

1. Post an issue in https://github.com/mikefarah/yq/issues to see if they have a suggestion on how to update the non object nodes below `workflows`.
1. Hacky solution: using an expression like ` yq r config.yml --printMode pv 'workflows.*.jobs.[*]', you get output like:
    ```
    workflows.build-and-test.jobs.[0]: test
    workflows.build-and-test.jobs.[1]:
      build:
        requires:
          - test
        filters:
          tags:
            only: /^v.*/
    workflows.build-and-test.jobs.[2]:
      deploy:
        requires:
          - build
        filters:
          tags:
            only: /^v.*/
          branches:
            ignore: /.*/
    workflows.frequently.jobs.[0]: test
    workflows.frequently.jobs.[1]:
      nuke_phx_devops:
        requires:
          - test
    workflows.nightly.jobs.[0]: test
    workflows.nightly.jobs.[1]:
      nuke_sandbox:
        requires:
          - test
    ```
    If you parse that with Go and pull the entries that have no children (e.g., `workflows.frequently.jobs.[0]: test`) into a list, you can loop over each item `x` in that list and do `yq w config.yml 'x.context[+]' 'Gruntwork Admin'` (e.g., `yq w config.yml 'workflows.build-and-test.jobs.[0].test.context[+]' 'Gruntwork Admin').

Reply:
Awesome- thanks, @brikis98  I'll get #1 cooking and in the meanwhile take a stab at playing around with option #2 as well.

Reply:
https://github.com/mikefarah/yq/issues/586

Reply:
Mike got back to us quickly and confirmed that this is not currently supported - the forthcoming `v4` rewrite of `yq` will support this and more - so I will forge ahead with an implementation of option 2.

Reply:
Roger!

Reply:
Fixed in https://github.com/gruntwork-io/prototypes/commit/11ebf9f18eb1b467544ba9831cf5c1f8fa69d510